### DISCUSSION TITLE
Can't find an example of apollo-server-micro (v3) using subscriptions

### DISCUSSION THREAD

**Discussion by gablabelle** [apollographql/apollo-server#6626] [Q&A]
Hello all,

I'm simply looking for an example of apollo-server-micro (v3) using subscriptions. 

Reading https://www.apollographql.com/docs/apollo-server/data/subscriptions#enabling-subscriptions

> Beginning in Apollo Server 3, subscriptions are not supported by the "batteries-included" apollo-server package. To enable subscriptions, you must first [swap to the apollo-server-express package](https://www.apollographql.com/docs/apollo-server/integrations/middleware/#swapping-out-apollo-server) (or any other Apollo Server integration package that supports subscriptions).

So does that mean that using `apollo-server-express` is mandatory for me to access to the configuration bits to get subscriptions working?

I wish I could leverage the next.js server to run my apollo server instead of adding a new express server instance to my container. I think [this take by @theobr](https://www.youtube.com/watch?v=2cB5Fh46Vi4) nicely explains why.

**Comment by gablabelle** [comment]
What I think I'll do is probably use a custom next.js server as described [here](https://nextjs.org/docs/advanced-features/custom-server) & [here](https://github.com/vercel/next.js/blob/canary/examples/custom-server-express/server.js). That way, I'll be able to hook into the server and avoid having to run 2 servers.

**Comment by glasser** [comment]
next.js does not actually have micro built into it. There is no technical relationship between next.js and micro, and it's very frustrating that for no obvious reason all of the next.js examples are built using the least functional Apollo Server flavor. You can easily use next.js to run express apps, as shown in its own docs. Perhaps one day they will look at our PR to fix their examples: https://github.com/vercel/next.js/pull/30082
